RefrigeratorExe/RefrigeratorExe/Refrigerator.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace RefrigeratorExe
+{
+    internal class Refrigerator :IComparable<Refrigerator>
+    {
+        private string _model;
+
+
+        public Guid id { get; }
+        public string Model
+        {
+            get
+            {
+                return _model;
+            }
+            private set
+            {
+                if (!value.Equals(""))
+                    _model = value;
+                else
+                    throw new ArithmeticException("Illegal model ,model must contain a String ");
+            }
+        }
+        public string Color { get; set; }
+        public int NumberOfShelfs { get; set; }
+        public List<Shelf> Shelves;
+
+
+        public Refrigerator(string model, string color, int shelf)
+        {
```

Comment:
constructor body should first validate the arguments and then set the member properties:

public Refrigerator(string model, string color, int shelfCount)
{
    if (model is null or "")
        throw new ArgumentException("Model cannot be empty");
    if (model.Length > MaxModelLength)
        throw new ArgumentException("Model name length exceeds maximum value", $"{MaxModelLength}");

    Model = model;
    Id = Guid.NewGuid();
    Color = color;
    Shelves = new List<Shelf>(shelfCount);
}